pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -0,0 +1,625 @@
+    optional string name = 2;
+    // Reference to the image in use. For most runtimes, this should be an
+    // image ID.
+    optional string image_ref = 3;
+    // State is the state of the container.
+    optional ContainerState state = 4;
+    // Labels are key value pairs that may be used to scope and select individual resources.
+    map<string, string> labels = 5;
+}
+
+message ListContainersResponse {
+    // List of containers
+    repeated Container containers = 1;
+}
+
+message ContainerStatusRequest {
+    // The id of the container
+    optional string container_id = 1;
+}
+
+// ContainerStatus represents the status of a container.
+message ContainerStatus {
+    // ID of the container.
+    optional string id = 1;
+    // Name of the container.
+    optional string name = 2;
+    // Status of the container.
+    optional ContainerState state = 3;
+    // Creation time of the container.
+    optional int64 created_at = 4;
+    // Start time of the container.
+    optional int64 started_at = 5;
+    // Finish time of the container.
+    optional int64 finished_at = 6;
+    // Exit code of the container.
+    optional int32 exit_code = 7;
+    // Reference to the image in use. For most runtimes, this should be an
+    // image ID
+    optional string image_ref = 8; 
+    // A string explains why container is in such a status.
+    optional string reason = 9;
+    // Labels are key value pairs that may be used to scope and select individual resources.
+    map<string,string> labels = 10;
+    // Annotations is an unstructured key value map.
+    map<string,string> annotations = 11;
+    // Mounts specifies mounts for the container
+    repeated Mount mounts = 12;
+}
+
+message ContainerStatusResponse {
+    // The status of the container
+    optional ContainerStatus status = 1;
+}
+
+message ExecRequest {
+    // The id of the container
+    optional string container_id = 1;
+    // The cmd to execute
+    repeated string cmd = 2;
+    // Whether use tty
+    optional bool tty = 3;
+    // Streaming stdin
+    optional bytes stdin = 4;
+}
+
+message ExecResponse {
+    // Streaming stdout
+    optional bytes stdout = 1;
+    // Streaming stderr
+    optional bytes stderr = 2;
+}
+
+message ImageFilter {
+    // The spec of the image
+    optional ImageSpec image = 1;
+}
+
+message ListImagesRequest {
+    // The filter to list images
+    optional ImageFilter filter = 1;
+}
+
+// Basic information about a container image.
+message Image {
+    // ID of the image.
+    optional string id = 1;
+    // Other names by which this image is known.
+    repeated string repo_tags = 2;
+    // Digests by which this image is known.
+    repeated string repo_digests = 3;
+    // The size of the image in bytes.
+    optional uint64 size = 4;
+}
+
+message ListImagesResponse {
+    // List of images
+    repeated Image images = 1;
+}
+
+message ImageStatusRequest {
+    // The spec of the image
+    optional ImageSpec image = 1;
```

Comment:
Probably replace `ImageSpec` with `Image`, the former one only contains a string.
Or just keep one of them?